airbyte-ci/connectors/pipelines/pipelines/builds/java_connectors.py

```diff
@@ -11,36 +11,42 @@
 
 
 class BuildConnectorDistributionTar(GradleTask):
+    """
+    A step to build a Java connector image using the distTar Gradle task.
+    """
+
     title = "Build connector tar"
     gradle_task_name = "distTar"
 
+
     async def _run(self) -> StepResult:
-        cdk_includes = ["./airbyte-cdk/java/airbyte-cdk/**"]
-        with_built_tar = (
-            environments.with_gradle(
-                self.context,
-                self.build_include + cdk_includes,
-            )
-            .with_exec(["./gradlew", ":airbyte-cdk:java:airbyte-cdk:publishSnapshotIfNeeded"])
-            .with_mounted_directory(str(self.context.connector.code_directory), await self.context.get_connector_dir())
-            .with_exec(self._get_gradle_command())
-            .with_workdir(f"{self.context.connector.code_directory}/build/distributions")
-        )
+        result = await super()._run()
+        if result.status is not StepStatus.SUCCESS:
+            return result
+
+        with_built_tar = result.output_artifact.with_workdir(f"{self.context.connector.code_directory}/build/distributions")
```

Comment:
Do you like this `result.output_artifact` pattern? I originally find it was a good idea to make step return whatever artifact they like following their execution. But I think that it impedes readability. I'd now prefer a default : `result.container`. Wdyt? (not to change in the scope of this PR)

Reply:
Honestly, I don't have an opinion, as I'm not familiar with the usage in the rest of the airbyte-ci codebase.